### DISCUSSION TITLE
MorphMany withTrashed is not working

### DISCUSSION THREAD

**Discussion by yankewei** [laravel/framework#35976] [Q&A]
- Laravel Version: 8.21.0
- PHP Version: 7.4.9
- Database Driver & Version: Mysql & 8.0.21

### Description:
I have three tables on db.
```
-  comments
    - id
    - body
    - commentable_id
    - commentable_type
    - deleted_at

- posts
    - id
    - title
    - deleted_at

- video
    - id
    - title
    - url
```
These three tables correspond to three models.
```php
class Comment extends Model
{
    use SoftDeletes;

    public function commentable()
    {
        return $this->morphTo();
    }
}

class Post extends Model
{
    use SoftDeletes;
    
    public function comments()
    {
        return $this->morphMany('App\Models\Comment', 'commentable');
    }

}

class Video extends Model
{

    public function comments()
    {
        return $this->morphMany('App\Models\Comment', 'commentable');
    }
}
```
Now I want to get the comment, and the entity corresponding to the comment. Notice, **comment model is using soft delete**
```php
Comment::with(['commentable' => function (MorphTo $query) {
    return $query->withTrashed();
}])->get();

// This working fine.
```

Then, comment does not use soft delete. 
```php
class Comment extends Model
{
    // use SoftDeletes;

    public function commentable()
    {
        return $this->morphTo();
    }
}
```
I got an exception:
`BadMethodCallException : Call to undefined method Illuminate\Database\Eloquent\Builder::withTrashed()`, it look like get video entity is failed.

**Comment by dyaskur** [comment]


**Reply by yankewei** [reply]
As long as the comment Model uses `SoftDeletes`, post and video can work regardless of whether `SoftDeletes` is used or not.

**Comment by dyaskur** [comment]
This way should work:

```
Comment::with(['commentable' => function (MorphTo $query) {
try {
    return $query->withTrashed();
} catch (\Exception $e) {
    return $query;
}}])->get()
```

reference: https://laracasts.com/discuss/channels/laravel/softdelete-on-polymorphic-relationships

**Reply by yankewei** [reply]
I know how to solve this problem, but i think this is a bug.
 It should not working because `Video` Model has not use  `SoftDeletes`, but as long as `Comment` Model uses `SoftDeletes`, everything is working fine.
